File: r/boot.R
# ///▙▖▙▖▞▞▙▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂
BANNER <- "///▙▖▙▖▞▞▙▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂"
lawline <- function() cat(BANNER, "\n")

# Token classification system
classify <- function(s) {
  if (grepl("//▞▞", s)) return("IMPRINT")
  if (grepl("▛///▞", s)) return("SECTION_HEAD")
  if (grepl("▞▞//▟", s)) return("SECTION_TAIL")
  if (grepl("▛▞\\s+.+?\\s+▞//", s)) return("NAMED_SECTION")
  if (grepl("▞▞", s)) return("COLON2")
  if (grepl("//▞", s)) return("OPEN_MINOR")
  "TEXT"
}

extract_section_name <- function(s) {
  if (grepl("▛▞\\s+.+?\\s+▞//", s)) {
    sub(".*▛▞\\s+(.+?)\\s+▞//.*", "\\1", s)
  } else {
    NA
  }
}

File: ruby/boot.rb
# ///▙▖▙▖▞▞▙▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂
BANNER = "///▙▖▙▖▞▞▙▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂"

def lawline
  puts BANNER
end

# Token classification system
TOKENS = {
  %r{//▞▞} => :IMPRINT,
  %r{▛///▞} => :SECTION_HEAD,
  %r{▞▞//▟} => :SECTION_TAIL,
  %r{▛▞\s+(.+?)\s+▞//} => :NAMED_SECTION,
  %r{▞▞} => :COLON2,
  %r{//▞} => :OPEN_MINOR,
}

def classify(line)
  TOKENS.each { |rx, tag| return tag if line.match?(rx) }
  :TEXT
end

def extract_section_name(line)
  match = line.match(/▛▞\s+(.+?)\s+▞\/\//)
  match ? match[1] : nil
end
